### PR TITLE
plugins/zellij-nvim: init

### PR DESCRIPTION
**Pull request by hmajid2301**
Add support for [zellij-nvim](https://github.com/Lilja/zellij.nvim).

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Thank you for your contribution !
Please, use the new `mkNeovimPlugin` helper. To help you with the refactor, have a look at the `TEMPLATE.nix` file and also `grep` the codebase for `mkNeovimPlugin`.

If you need some help, don't hesitate to ask in the _Development_ matrix channel.

File: plugins/utils/zellij.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.zellij;
+in {
+  meta.maintainers = [maintainers.GaetanLepage];
```
Comment (GaetanLepage) [inline plugins/utils/zellij.nix:11]: Why not adding yourself as the maintainer ?

File: plugins/utils/zellij.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "zellij-nvim";
```
Comment (GaetanLepage) [inline plugins/utils/zellij.nix:9]:
```suggestion
  name = "zellij";
```

This is the name that the nixvim module will have, `zellij` in this case.

File: plugins/utils/zellij.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  originalName = "zellij.nvim";
+
+  defaultPackage = pkgs.vimPlugins.my-plugin-nvim;
```
Comment (GaetanLepage) [inline plugins/utils/zellij.nix:13]:
```suggestion
  defaultPackage = pkgs.vimPlugins.zellij-nvim;
```

File: plugins/utils/zellij.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    '';
+
+    vimTmuxNavigatorKeybinds = helpers.defaultNullOpts.mkBool false ''
+      Will set keybinds like <C-h> to left
+    '';
+
+    debug = helpers.defaultNullOpts.mkBool false ''
+      Will log things to /tmp/zellij.nvim
+    '';
+  };
```
Comment (GaetanLepage) [inline plugins/utils/zellij.nix:31]: Add a `settingsExample` with the different settings set to some value (could be the default ones).

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

**Comment by hmajid2301** [comment]
Thank you for the swift review. I am going to put this PR in draft until the lock file is updated because i think the zellj-nvim package is not yet available.

Also the new nvim nix helper, was really cool and easy to use :).

File: plugins/utils/zellij.nix (hunk 4 above)
Comment (hmajid2301) [reply plugins/utils/zellij.nix:31]: done

File: plugins/utils/zellij.nix (hunk 3 above)
Comment (hmajid2301) [reply plugins/utils/zellij.nix:13]: done

File: plugins/utils/zellij.nix (hunk 2 above)
Comment (hmajid2301) [reply plugins/utils/zellij.nix:9]: done

File: plugins/utils/zellij.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "zellij";
+
+  originalName = "zellij-nvim";
```
Comment (GaetanLepage) [inline plugins/utils/zellij.nix:11]:
```suggestion
  originalName = "zellij.nvim";
```

File: plugins/utils/zellij.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      Will set keybinds like <C-h> to left
+    '';
+
+    debug = helpers.defaultNullOpts.mkBool false ''
+      Will log things to /tmp/zellij.nvim
+    '';
+  };
+
+  settingsExample = {
+    enable = true;
```
Comment (GaetanLepage) [inline plugins/utils/zellij.nix:34]:
```suggestion
```

`enable` is not a `settings` option.

**Comment by GaetanLepage** [comment]
> Thank you for the swift review. I am going to put this PR in draft until the lock file is updated because i think the zellj-nvim package is not yet available.
> 
> Also the new nvim nix helper, was really cool and easy to use :).

You're welcome !
The lockfile should be updated very soon (#1329).

**Comment by hmajid2301** [comment]
Apologies for the sloppy PR I was fixing them as you suggest them! I just have a habit of committing at random.

**Review by GaetanLepage — APPROVED** [review]

**Pull request merged**